gradle/proguard-rules.pro

```diff
@@ -73,14 +73,13 @@
 # ZoneRulesProvider _does_ exist!
 -dontwarn java.time.zone.ZoneRulesProvider
 
-# Arouter
--keep public class com.alibaba.android.arouter.routes.**{*;}
--keep public class com.alibaba.android.arouter.facade.**{*;}
--keep class * implements com.alibaba.android.arouter.facade.template.ISyringe{*;}
--keep interface * implements com.alibaba.android.arouter.facade.template.IProvider
--keep class * implements com.alibaba.android.arouter.facade.template.IProvider
-
 # DataBinding
 -keep public class * extends androidx.databinding.ViewDataBinding {
   * inflate(android.view.LayoutInflater);
 }
+
+# Extra rules for R8 fullMode
+-keep,allowobfuscation class io.goooler.demoapp.common.base.binding.BaseBindingActivity
+-keep,allowshrinking class * extends io.goooler.demoapp.common.base.binding.BaseBindingActivity
```

Comment:
- `keep,allowobfuscation class BaseBindingActivity` for subclasses extends `BaseBindingActivity` directly.
- `-keep,allowshrinking class * extends BaseBindingActivity` for subclass use with Hilt. (it works after #31)